{{cookiecutter.project_name}}/pyproject.toml

```
@@ -12,16 +12,19 @@ requires = [
 ]
 build-backend = "{{'poetry.core.masonry.api' if cookiecutter.dependency_manager != 'pip' else 'setuptools.build_meta'}}"
 {{'\n[project.license]\nfile = "LICENSE"\n' if cookiecutter.dependency_manager == 'pip'}}
+
 {{'[tool.poetry.group.dev.dependencies]' if cookiecutter.dependency_manager != 'pip' else '[project.optional-dependencies]\ndev = ['}}
-{{'pytest = ">=8.1.1"' if cookiecutter.dependency_manager != 'pip' else ' "pytest==8.1.1",'}}
-{{'pytest-cov = ">=4.1.0"' if cookiecutter.dependency_manager != 'pip' else '    "pytest-cov==4.1.0",'}}
-{{'pre-commit = ">=3.6.2"' if cookiecutter.dependency_manager != 'pip' else '    "pre-commit==3.6.2",'}}
-{{'pdoc = ">=14.1.0"' if cookiecutter.dependency_manager != 'pip' else '    "pdoc==14.1.0",'}}
-{{'' if cookiecutter.dependency_manager != 'pip' else '    "coverage[toml]==5.5",'}}{{'readme-coverage-badger = ">=0.1.2"' if cookiecutter.dependency_manager != 'pip' else '\n    "readme-coverage-badger==0.1.2",'}}
-{{'click = ">=8.1.7"' if cookiecutter.dependency_manager != 'pip' else '    "click==8.1.7",\n]'}}
-{{'ruff = ">=0.3.2"' if cookiecutter.dependency_manager != 'pip' else '    "ruff==0.3.2",\n]'}}
-{{'mypy = ">=1.9.0"' if cookiecutter.dependency_manager != 'pip' else '    "mypy==1.9.0",\n]'}}
-{{'nbstripout = ">=0.7.1"' if cookiecutter.dependency_manager != 'pip' else ' "nbstripout==0.7.1",\n]'}}
+{{'pytest = ">=8.1.1"'     if cookiecutter.dependency_manager != 'pip' else ' "pytest==8.1.1",'}}
+{{'pytest-cov = ">=4.1.0"' if cookiecutter.dependency_manager != 'pip' else ' "pytest-cov==4.1.0",'}}
+{{'pre-commit = ">=3.6.2"' if cookiecutter.dependency_manager != 'pip' else ' "pre-commit==3.6.2",'}}
+{{'pdoc = ">=14.1.0"'      if cookiecutter.dependency_manager != 'pip' else ' "pdoc==14.1.0",'}}
+{{'readme-coverage-badger = ">=0.1.2"' if cookiecutter.dependency_manager != 'pip' else ' "readme-coverage-badger==0.1.2",'}}
+{{'click = ">=8.1.7"'      if cookiecutter.dependency_manager != 'pip' else ' "click==8.1.7",'}}
+{{'ruff = ">=0.3.2"'       if cookiecutter.dependency_manager != 'pip' else ' "ruff==0.3.2",'}}
+{{'mypy = ">=1.9.0"'       if cookiecutter.dependency_manager != 'pip' else ' "mypy==1.9.0",'}}
+{{'nbstripout = ">=0.7.1"' if cookiecutter.dependency_manager != 'pip' else ' "nbstripout==0.7.1",'}}
+{{''                       if cookiecutter.dependency_manager != 'pip' else ' "coverage[toml]==5.5",'}}
+{{'\n' if cookiecutter.dependency_manager != 'pip' else '\n]'}}
 {{'\n[[tool.poetry.source]]\nname = "pypi"' if cookiecutter.dependency_manager != 'pip'}}
```

For Poetry projects, this creates 3 superfluous blank lines between spaces. For Pip projects, it creates a superfluous blank line at the end of the dependency list, and a superfluous blank line after that list. This should fix both of those things:

```suggestion
{{''                       if cookiecutter.dependency_manager != 'pip' else ' "coverage[toml]==5.5",'}}
{{'[[tool.poetry.source]]\nname = "pypi" if cookiecutter.dependency_manager != 'pip' else ']'}}
```